### PR TITLE
[LS] Add check text edit generation and fix return type support for methods in services/classes

### LINKED ISSUE #29312: The `Create Variable and Check` code action not working
**Description:**
Consider the following source snippet

```
import ballerina/observe;
import ballerina/http;

service / on new http:Listener(9234) {
    resource function get getResource(http:Caller caller, http:Request req) {
        http:Response res = new;
        observe:startSpan("MyFirstLogic");
    }
}

```

The code action in $title not working for `observe:startSpan("MyFirstLogic");`

Code action should generate the `int startSpan = check observe:startSpan("MyFirstLogic");` and instead generate int|error startSpan = observe:startSpan("MyFirstLogic");

**Affected Versions:**
sl-alpha3-SNAPSHOT at least

### LINKED ISSUE #29313: Modify Type Guard code action's template
**Description:**
Remove the parentheses around the if statement's expression generated from the `typeguard code action`

**Affected Versions:**
slp-alpha3-snapshot at least

### LINKED ISSUE #29314: NPE thrown from FixReturnTypeCodeAction
**Description:**
NPE thrown from FixReturnTypeCodeAction time to time. Possible NPE in https://github.com/ballerina-platform/ballerina-lang/blob/master/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/FixReturnTypeCodeAction.java#L133

```
"Error: java.lang.NullPointerException\n\tat org.ballerinalang.langserver.codeaction.providers.changetype.FixReturnTypeCodeAction.getFunctionNode(FixReturnTypeCodeAction.java:133)"
```

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by IMS94**
## Purpose
$subject

Fixes #29312
Fixes #29314
Fixes #29313

## Approach
Updated `CodeActionUtil` to check for resource methods and methods and extract the function definition node.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java [hunk 1]
```diff
@@ -752,20 +761,40 @@ private static boolean isWithinRange(int positionOffset, int startOffSet, int en
...
+                functionDefNode = (FunctionDefinitionNode) parentNode;
+                break;
+            } else if (parentNode.kind() == SyntaxKind.RESOURCE_ACCESSOR_DEFINITION &&
+                    parentNode.parent().kind() == SyntaxKind.SERVICE_DECLARATION) {
                 functionDefNode = (FunctionDefinitionNode) parentNode;
                 break;
             }
+
+            parentNode = parentNode.parent();
         }
```
Comment (pubudu91) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java:796]: Don't we have to consider method declarations here?

**Review by nadeeshaan — CHANGES_REQUESTED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java [hunk 2]
```diff
@@ -760,12 +769,27 @@ private static boolean isWithinRange(int positionOffset, int startOffSet, int en
             if (parentNode == null) {
                 break;
             }
+
+            if (parentNode.parent() == null) {
+                break;
+            }
```
[the remark this answers is not in the document]
Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java]: @nadeeshaan this is outdated. I have changed the logic here.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java (hunk 1 above)
Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java:796]: In this method we are interested in functions with a body (to fix return types, add `returns error?` when adding `check`s via code actions). Therefore, we don't have to consider method declarations here IMo.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java [hunk 3]
```diff
@@ -752,33 +761,53 @@ private static boolean isWithinRange(int positionOffset, int startOffSet, int en
...
+                break;
+            } else if (parentNode.kind() == SyntaxKind.OBJECT_METHOD_DEFINITION &&
+                    parentNode.parent().kind() == SyntaxKind.CLASS_DEFINITION) {
+                functionDefNode = (FunctionDefinitionNode) parentNode;
+                break;
+            } else if (parentNode.kind() == SyntaxKind.RESOURCE_ACCESSOR_DEFINITION &&
+                    parentNode.parent().kind() == SyntaxKind.SERVICE_DECLARATION) {
                 functionDefNode = (FunctionDefinitionNode) parentNode;
                 break;
             }
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java:793]: We can combine all these cases to single if clause and also let's move the if clause's check to a separate private util.

**Review by nadeeshaan — CHANGES_REQUESTED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java (hunk 3 above)
Comment (IMS94) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java]: Done. Can you check now?

**Review by nadeeshaan — APPROVED** [review]

**Pull request merged**